extensions.html

```diff
@@ -354,18 +355,19 @@ <h2>
             [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
             </li>
             <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-            is not `undefined`, [=queue a global task=] on the [=relevant
-            global object=] of [=this=] using the [=gamepad task source=] to
-            run the following steps:
+            is not `undefined`:
               <ol>
-                <li>[=Resolve=]
-                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-                with {{GamepadHapticsResult/"preempted"}}.
+                <li>Let |effectPromise| be
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
                 </li>
                 <li>Set
                 [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
                 `undefined`.
                 </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
```

Comment:
As above?? 

```suggestion
                </li>
                <li>Terminate this algorithm.</li>
```

Reply:
If we terminate here then we won't stop vibration when there's an effect playing.